power_grid_model_io_native_c/power_grid_model_io_native_c/src/vnf_pgm_converter.cpp

```diff
@@ -12,21 +13,29 @@
 
 #include <power_grid_model/auxiliary/dataset.hpp>
 
-using power_grid_model::ConstDataset;
+using namespace power_grid_model_io_native;
 
 struct PGM_IO_VnfConverter : public PgmVnfConverter {
     using PgmVnfConverter::PgmVnfConverter;
 };
 
-// TODO(Laurynas-Jagutis) add call_with_catch for these functions.
-PGM_IO_VnfConverter* PGM_IO_create_vnf_converter(const PGM_IO_Handle* /*handle*/, char* file_buffer) {
-    auto* converter = new PGM_IO_VnfConverter(file_buffer);
-    parse_vnf_file_wrapper(converter);
-    return converter;
+PGM_IO_VnfConverter* PGM_IO_create_vnf_converter(PGM_IO_Handle* handle, char const* file_buffer,
+                                                 PGM_IO_ExperimentalFeatures experimental_features) {
+    return call_with_catch(
+        handle,
+        [file_buffer, experimental_features] {
+            auto const experimental_feature =
+                static_cast<ExperimentalFeatures>(static_cast<IntS>(experimental_features));
```

Comment:
please verify whether this is also how we do it in the PGM C API. if so, great. if not, please homogenize

Reply:
for example in serialization.cpp, we do:
`static_cast<power_grid_model::SerializationFormat>(serialization_format)` where serialization_format is a PGM_Idx and not really an enum like we have in my case

Reply:
if that is true then please look at one of the actual enum types. i think we use `switch`/`case` statements for that, right?